providers/default.rb

```diff
@@ -1,16 +1,30 @@
 use_inline_resources
 
 action :update do
+  dir        = new_resource.base_name.split('/')[0..-2].join('/')
+  push_event = new_resource.push_to_datadog || false
+  event_name = new_resource.datadog_event_name
+  event_text = new_resource.datadog_event_text
+  event_key  = new_resource.datadog_event_key
+
```

Comment:
Why do you introduce all these variables instead of using `new_resource.<attribute_name>`?